commit 60a17df4c78d7bd9f27ca1425dba6c53298fd3d3
Author: Aleksander Morgado <aleksander@aleksander.es>
Date:   Fri Feb 6 15:55:18 2015 +0100

    novatel: build a common non-inst library with the core Novatel modem support

---
 plugins/Makefile.am |   22 ++++++++++++++++------
 1 file changed, 16 insertions(+), 6 deletions(-)

Index: b/plugins/Makefile.am
===================================================================
--- a/plugins/Makefile.am
+++ b/plugins/Makefile.am
@@ -442,6 +442,19 @@ libmm_plugin_iridium_la_SOURCES = \
 libmm_plugin_iridium_la_CPPFLAGS = $(PLUGIN_COMMON_COMPILER_FLAGS)
 libmm_plugin_iridium_la_LDFLAGS = $(PLUGIN_COMMON_LINKER_FLAGS)
 
+# Common Novatel modem support library
+noinst_LTLIBRARIES += libmm-utils-novatel.la
+libmm_utils_novatel_la_SOURCES = \
+	novatel/mm-common-novatel.c \
+	novatel/mm-common-novatel.h \
+	novatel/mm-broadband-modem-novatel.c \
+	novatel/mm-broadband-modem-novatel.h
+libmm_utils_novatel_la_CPPFLAGS = $(PLUGIN_COMMON_COMPILER_FLAGS)
+libmm_utils_novatel_la_LIBADD = $(GUDEV_LIBS) $(MM_LIBS)
+
+NOVATEL_COMMON_COMPILER_FLAGS = -I$(top_srcdir)/plugins/novatel
+NOVATEL_COMMON_LIBADD_FLAGS = $(builddir)/libmm-utils-novatel.la
+
 # Novatel LTE modem
 libmm_plugin_novatel_lte_la_SOURCES = \
 	novatel/mm-plugin-novatel-lte.c \
@@ -458,13 +471,10 @@ libmm_plugin_novatel_lte_la_LDFLAGS = $(
 # Novatel non-LTE modem
 libmm_plugin_novatel_la_SOURCES = \
 	novatel/mm-plugin-novatel.c \
-	novatel/mm-plugin-novatel.h \
-	novatel/mm-common-novatel.c \
-	novatel/mm-common-novatel.h \
-	novatel/mm-broadband-modem-novatel.c \
-	novatel/mm-broadband-modem-novatel.h
-libmm_plugin_novatel_la_CPPFLAGS = $(PLUGIN_COMMON_COMPILER_FLAGS)
+	novatel/mm-plugin-novatel.h
+libmm_plugin_novatel_la_CPPFLAGS = $(PLUGIN_COMMON_COMPILER_FLAGS) $(NOVATEL_COMMON_COMPILER_FLAGS)
 libmm_plugin_novatel_la_LDFLAGS = $(PLUGIN_COMMON_LINKER_FLAGS)
+libmm_plugin_novatel_la_LIBADD = $(NOVATEL_COMMON_LIBADD_FLAGS)
 
 # Altair LTE modem
 libmm_plugin_altair_lte_la_SOURCES = \
